model/src/payout_curve/quanto.rs

```diff
@@ -230,19 +231,56 @@ impl Curve {
     /// Price interval at which the party going long gets liquidated i.e. their payout amount equals
     /// zero.
     fn long_liquidation_interval(&self) -> Result<RangeInclusive<u64>> {
-        let bankruptcy_price = bankruptcy_price_long(self.initial_price, self.leverage_long);
+        let initial_margin = self
+            .initial_margin_long()
+            .to_signed()
+            .context("Could not convert long's initial margin to bitcoin::SignedAmount")?;
+        let effective_initial_margin =
+            initial_margin + self.fee_offset.as_signed_amount(Position::Long);
+        let effective_initial_margin = effective_initial_margin
+            .to_unsigned()
+            .context("Could not convert long's effective initial margin to bitcoin::Amount")?;
+
+        let effective_leverage = self.effective_leverage(effective_initial_margin)?;
+        let bankruptcy_price = bankruptcy_price_long(self.initial_price, effective_leverage);
 
         Ok(0..=bankruptcy_price)
     }
 
     /// Price interval at which the party going short gets liquidated i.e. their payout amount
     /// equals zero.
     fn short_liquidation_interval(&self) -> Result<RangeInclusive<u64>> {
-        let bankruptcy_price = bankruptcy_price_short(self.initial_price, self.leverage_short);
+        let initial_margin = self
+            .initial_margin_short()
+            .to_signed()
+            .context("Could not convert short's initial margin to bitcoin::SignedAmount")?;
+        let effective_initial_margin =
+            initial_margin + self.fee_offset.as_signed_amount(Position::Short);
+        let effective_initial_margin = effective_initial_margin
+            .to_unsigned()
+            .context("Could not convert short's effective initial margin to bitcoin::Amount")?;
+
+        let effective_leverage = self.effective_leverage(effective_initial_margin)?;
+        let bankruptcy_price = bankruptcy_price_short(self.initial_price, effective_leverage);
 
         Ok(bankruptcy_price..=maia_core::interval::MAX_PRICE_DEC)
     }
 
+    /// Computes effective leverage based on updated margin
+    ///
+    /// For calculating the liquidation price we need to take into account the potentially updated
+    /// margin (reduced or increased due to fee account). The formula below was derived from the
+    /// formula to compute the initial margin in the first place: `margin=(n_contracts *
+    /// initial_price * multiplier) / leverage` --> `effective_leverage=(n_contracts *
+    /// initial_price * multiplier) / margin`
+    fn effective_leverage(&self, margin: Amount) -> Result<Decimal> {
+        let effective_leverage =
+            (Decimal::from(self.n_contracts) * Decimal::from(self.initial_price) * self.multiplier)
+                / Decimal::from_f64(margin.as_btc())
+                    .context("Could not convert initial margin to decimal")?;
+        Ok(effective_leverage)
```

Comment:
🙃 If you put the formula on a separate line from all the conversions you probably don't need to add it to the documentation.